cranelift/codegen/src/isa/riscv64/inst/vector.rs

```diff
@@ -245,27 +246,46 @@ impl VecAluOpRRR {
         0x57
     }
     pub fn funct3(&self) -> u32 {
-        match self {
-            VecAluOpRRR::Vadd
-            | VecAluOpRRR::Vsub
-            | VecAluOpRRR::Vand
-            | VecAluOpRRR::Vor
-            | VecAluOpRRR::Vxor => VecOpCategory::OPIVV,
-            VecAluOpRRR::Vmul | VecAluOpRRR::Vmulh | VecAluOpRRR::Vmulhu => VecOpCategory::OPMVV,
-        }
-        .encode()
+        self.category().encode()
     }
     pub fn funct6(&self) -> u32 {
         // See: https://github.com/riscv/riscv-v-spec/blob/master/inst-table.adoc
         match self {
-            VecAluOpRRR::Vadd => 0b000000,
-            VecAluOpRRR::Vsub => 0b000010,
-            VecAluOpRRR::Vmul => 0b100101,
-            VecAluOpRRR::Vmulh => 0b100111,
-            VecAluOpRRR::Vmulhu => 0b100100,
-            VecAluOpRRR::Vand => 0b001001,
-            VecAluOpRRR::Vor => 0b001010,
-            VecAluOpRRR::Vxor => 0b001011,
+            VecAluOpRRR::VaddVV | VecAluOpRRR::VaddVX => 0b000000,
+            VecAluOpRRR::VsubVV | VecAluOpRRR::VsubVX => 0b000010,
```

Comment:
It's surprising that the vector/vector and vector/scalar combinations have the same `funct6` value. Does something else end up distinguishing those cases?

Reply:
Yes, `funct3` usually dictates the source/destination register types for most opcodes.